### PR TITLE
(gh-pages): Fixing script to conditionally generate docs

### PR DESCRIPTION
**Pull request by vinagesh**
# Checklist
- [ ] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [ ] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem

# Description of the solution
 

Here is an example of the pipeline run - https://github.com/Azure/azure-iot-sdk-java/tree/gh-pages-2020-12-09
I tweaked the pipeline to run off of my branch to get this. If you look in the contents of the branch, only one folder inside master is changed corresponding to the run (because only that was set to true)

### CODE REVIEW

File: vsts/generate_javadocs_branch.ps1 [hunk 1]
```diff
@@ -56,85 +56,76 @@ function CreateJavadocReleaseBranch(
...
-
-    # Move the generated content to the correct folder. The folder will be different for master and preview branches.
-    New-Item -Path ../$FolderName -ItemType Directory
+    Set-Location apidocs    
 
     if ($UpdateDepsDocs -eq "True")
     {
-        Move-Item -Force -Path ..\deps -Destination ..\$FolderName\deps
+        Remove-Item ..\$FolderName\deps -Force -Recurse
+        New-item  -Path ..\$FolderName\deps -ItemType Directory
```
Comment (barustum) [inline vsts/generate_javadocs_branch.ps1:64]: extra space bewteen New-item and -path. Or is this coding style convention to indent the parameters with the commands below?

**Review by barustum — APPROVED** [review]

File: vsts/generate_javadocs_branch.ps1 (hunk 1 above)
Comment (vinagesh) [reply vsts/generate_javadocs_branch.ps1:64]: No it is just extra space. My eyes are not as good with seeing spaces :P

**Pull request merged**